index.rst

```diff
@@ -4,9 +4,149 @@ File Formats and Layouts for Cell-based Coadds
+The overheads would be significant for the PSF images, which we expect to be 25-40 pixels on a side (2.5-6 KB uncompressed).
+
+Subimage reads would be similarly non-ideal but perhaps tolerable.
+Because each HDU is so small, it'd be plenty efficient to read full HDUs, but only those for the cells that overlap the region of interest.
+Seeking to the right HDUs (or requesting the appropriate byte ranges, in the object store case) is easily solved by putting a table of byte offsets in the primary HDU header, though this isn't something third-party FITS readers could leverage.
+That would make for a simple solution to the problem of doing subimage reads over object stores (including compression): we could use the address table to read the HDUs we are interested in in their entirety into a client-side memory location that looks like a full in-memory FITS file holding just those HDUs, and then delegate to CFITSIO's "memory file" interfaces to let it do the decompression.
+
+As in the binary table case, it's an open question whether we could get sufficiently good compression ratios if we are limited to compressing one cell at a time.
+
+Data Cubes
+----------
+
+TODO
+
+Exploded Images
+---------------
+
+TODO
+
+Stitched Images
+---------------
+
+TODO
+
+Hybrid Options
+--------------
+
+TODO
+
+
+Metadata Layouts
 ================
 
+Cell coadd metadata falls into three categories:
+
+- global information common to all cells (identifiers, WCS, grid structure);
+- per-cell information with a fixed schema (including coadded aperture corrections and wavelength-dependent throughput);
+- visit, detector and other IDs for the observations that contribute to each cell.
+
+While some global information will go into FITS headers (certainly the WCS and some identifiers), we do not want to assume that all global metadata can be neatly represented in the limited confines of a FITS header.
+A single-row binary table is another option, but we will likely instead adopt the approach recently proposed for other Rubin image data products on RFC-1030: embedding a JSON document as a byte array in a FITS extension HDU.
+
+A binary table with per-cell rows is a natural fit for the fixed-schema per-cell information, especially if the image data layout already involves a binary table with per-cell rows.
+But if we're embedding a JSON document in the FITS file anyway, it might make more sense to store this information in JSON as well; this will let us share code, documentation, and serialization for more complex objects with other Rubin image data products, and that includes sharing the machinery for managing schema changes schema documentation.
```

Comment:
Have you seen performance issues with JSON in FITS? 

We do this very rarely in DES and it is a potentially lossy operation, especially with respect to floating point numbers etc.

Reply:
You do have to be very careful with the floating point numbers (NaNs and Infs especially), but the libraries we're using (mostly Pydantic) seem to be pretty good about that.

The JSON-in-FITS stuff is extremely new and we don't have any benchmarks of that specifically.  But I don't think the combination raises any red flags, and we've been pretty happy with JSON serialization even in some very big data structures (not YAML, though, avoid YAML like the plague if you care about performance!).

Reply:
My 2 cents:  As an end user I would prefer binary table if the metadata has a "flat" (non hierarchical) structure and has FITS compatible data types.

Using binary blobs makes data exploration/browsing more difficult for anyone who is not using DM specific "data loaders" etc.